spring-shell-autoconfigure/src/main/java/org/springframework/shell/boot/JLineShellAutoConfiguration.java

```diff
@@ -38,9 +39,11 @@
  * @author Florent Biville
  */
 @Configuration(proxyBeanMethods = false)
+@ConditionalOnProperty(prefix = "spring.shell.jline-shell", value = "enabled", havingValue = "true", matchIfMissing = true)
```

Comment:
I think we don't need this setting?

Reply:
> I think we don't need this setting?

Personally, I think if user wanna implement all the Beans under this configuration, we can provide a toggle to do it  instead of  loading this empty @Configuration/Component.

Reply:
Having flag in this level doesn't make sense as those beans are currently used elsewhere. Enabled flags are usually used in higher level feature sets.

Reply:
> Having flag in this level doesn't make sense as those beans are currently used elsewhere. Enabled flags are usually used in higher level feature sets.

I see, gonna an update here.